vscode/src/extension.ts

```diff
@@ -41,9 +41,24 @@ import { createReferenceProvider } from "./references.js";
 import { activateTargetProfileStatusBarItem } from "./statusbar.js";
 import { initFileSystem } from "./memfs.js";
 import { getManifest, readFile, listDir } from "./projectSystem.js";
+import {
+  Logging,
+  initLogForwarder,
+  initOutputWindowLogger,
+} from "./logging.js";
+
+export async function activate(
+  context: vscode.ExtensionContext,
+): Promise<ExtensionApi> {
+  const api: ExtensionApi = {};
+
+  if (context.extensionMode !== vscode.ExtensionMode.Test) {
```

Comment:
nit: reverse the conditional blocks so that we don't have to negate the check.

Reply:
Ha, I had an internal battle between "the preferred code branch should come first" and "no inequality checks", and the first one won. I'll switch it though.